### PR TITLE
Solution and test for #56 find minimum number of coins

### PR DESCRIPTION
**Pull request by yjlim5**

### CODE REVIEW

File: solutions/56.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// coin denomination problem
+
```
Comment (songz) [inline solutions/56.js]: add comments like this guy: https://github.com/llipio/algorithms/pull/55/files

File: solutions/56.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * Returns the minimum number of coins required to make up the given total value
+ * Each coin denomination can be used as many times as needed.
+ * @param {number[]} arr - An array of coin denomination numbers
+ * @param {number} total - The total value 
+ * @returns {number} min - The mininum number of coins which add up to the toal value
+ */
+
+const solution = (arr, total) => {
+  if (total === 0) {
```
Comment (songz) [inline solutions/56.js]: you will always be giving a 1 coin.

File: solutions/56.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    return 0;
+  }
+
+  // set minimum number of coins to the total divided by smallest denomination
+  let min = Math.floor(total / Math.min(...arr));
+  arr.forEach((denom) => {
+    if (total - denom >= 0) {
```
Comment (songz) [inline solutions/56.js]: perhaps you don't need an if statement.

File: solutions/56.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+const solution = (arr, total) => {
+  if (total === 0) {
+    return 0;
+  }
+
+  // set minimum number of coins to the total divided by smallest denomination
+  let min = Math.floor(total / Math.min(...arr));
+  arr.forEach((denom) => {
```
Comment (songz) [inline solutions/56.js]: use a reduce function here: `https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce?v=control`

File: test/56.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const expect = require('chai').expect;
+const solution = require('../solutions/56.js').solution;
+// solution = require('../yourSolution').solution;
+
+describe('return smallest number of coins required to return the change', () => {
+  it('simplest case [1],total=1', () => {
```
Comment (songz) [inline test/56.js]: you have 2 identical test cases.

File: test/56.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    expect(solution([1,2],2)).to.equal(1);
+  });
+  it('simple case [1,2], total=4', () => {
+    expect(solution([1,2],4)).to.equal(2);
+  });
+  it('hard case [2,3,6,7], total=12', () => {
+    expect(solution([2,3,6,7],12)).to.equal(2);
+  });
+});
+
```
Comment (songz) [inline test/56.js]: could you also add a 0 case?

File: solutions/56.js (hunk 3 above)
Comment (yjlim5) [inline solutions/56.js]: We need it to eliminate candidate solutions which cannot make up the total.

File: solutions/56.js (hunk 2 above)
Comment (songz) [inline solutions/56.js]: `<= 0`, then you wouldn't even need `if` check in reducer

Comment (songz) [inline solutions/56.js]: nvm

Comment (yjlim5) [inline solutions/56.js]: Thanks. That works too.

**Pull request merged**